client/client.cpp

```diff
@@ -128,11 +128,13 @@ bool Client::connect(
     const std::string& hostname,
     int port,
     int timeoutMs /* = -1 */) {
-  return connect(new Connection(hostname, port, timeoutMs));
+  return connect(std::unique_ptr<Connection>(
+        new Connection(hostname, port, timeoutMs)));
```

Comment:
Hm, on second thought -- did you check what happens if you call `connect()` twice now? It would be bad if you'd get an exception from the Connection constructor...

Reply:
It's a private function... It's up to us to not do stupid things? I don't see a reason to make this incredibly robust, since it was just a way to factor out some code.

Reply:
Well, the two `connect(Args...)` methods are public. The implementation changed in a way that a new Connection (which is RAII, i.e. it creates a zmq context, opens the socket and so on...) will be opened even if there is an existing one. Anyway, I quickly double-checked and calling connect() with an already active connection will not throw an error on the zmq side in Connection's constructor (which would be bad, since then we would have changed the interface from never exception -> sometimes exception).